rust/azure_iot_operations_protocol/src/telemetry/receiver.rs

```diff
@@ -218,6 +218,107 @@ pub struct Message<T: PayloadSerialize> {
     pub timestamp: Option<HybridLogicalClock>,
     /// Resolved static and dynamic topic tokens from the incoming message's topic.
     pub topic_tokens: HashMap<String, String>,
+    /// Incoming message topic
+    pub topic: String,
```

Comment:
This could be removed when the ChunkBuffer is implemented. There are other alternate ways we could get around adding this here if it's not desirable. However, it seemed a reasonable inclusion and made things much simpler.